packages/ovault-evm/contracts/OVaultComposer.sol

```diff
@@ -159,53 +151,77 @@ contract OVaultComposer is IOVaultComposer, ReentrancyGuard {
     /// @dev Permissionless function to send back the message to the source chain
     /// @dev Always possible unless the lzCompose() fails due to an Out-Of-Gas panic
     function refund(bytes32 _guid, bytes calldata _extraOptions) external payable nonReentrant {
+        if (failedGuidState(_guid) != FailedState.CanOnlyRefund) revert CanNotRefund(_guid);
+
         FailedMessage memory failedMessage = failedMessages[_guid];
+        delete failedMessages[_guid];
+
         SendParam memory refundSendParam = failedMessage.refundSendParam;
-        if (failedGuidState(_guid) != FailedState.CanOnlyRefund) revert CanNotRefund(_guid);
+        address refundOft = failedMessage.oft;
```

Comment:
@shankars99 it this ok?
Previousely this was `failedMessage.refundOFT` and now `failedMessage.oft` is used

In the case of `FailedState.CanOnlyRefund`: `failedMessage.oft == address(0))`, see:

https://github.com/LayerZero-Labs/devtools/blob/f4f5703eb7cd2b7280d9f69ce2c76ad7b8b3a073/packages/ovault-evm/contracts/OVaultComposer.sol#L262-L264